knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -154,75 +153,52 @@ func Delete(instance *servingv1alpha1.KnativeServing, apiclient client.Client, s
-		return fmt.Errorf("failed to delete kn ConsoleCLIDownload resources manifest: %w", err)
+	log.Info("Deleting kn ConsoleCLIDownload Service")
+	if err := apiclient.Delete(context.TODO(), makeKnService("", instance)); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete kn ConsoleCLIDownload Service: %w", err)
 	}
 
 	return nil
 }
 
-// manifest returns kn ConsoleCLIDownload deploymnet resources manifest after traformation
-func manifest(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) (mf.Manifest, error) {
-	manifest, err := RawManifest(apiclient)
-	if err != nil {
-		return mf.Manifest{}, fmt.Errorf("failed to read kn ConsoleCLIDownload deployment manifest: %w", err)
-	}
-
-	// 1. Use instance's namespace to deploy download resources into
-	// 2. Set proper kn-cli-artifacts image
-	// 3. Set Owner annotations
-	transforms := []mf.Transformer{mf.InjectNamespace(instance.GetNamespace()),
-		replaceKnCLIArtifactsImage(os.Getenv("IMAGE_KN_CLI_ARTIFACTS"), scheme),
-		common.SetOwnerAnnotations(instance),
-	}
-
-	manifest, err = manifest.Transform(transforms...)
-	if err != nil {
-		return mf.Manifest{}, fmt.Errorf("failed to transform kn ConsoleCLIDownload resources manifest: %w", err)
-	}
-
-	return manifest, nil
-}
-
-// manifest returns kn ConsoleCLIDownload deploymnet resources manifest without transformation
-func RawManifest(apiclient client.Client) (mf.Manifest, error) {
-	return mfc.NewManifest(manifestPath(), apiclient, mf.UseLogger(log.WithName("mf")))
-}
-
-// manifestPath returns kn ConsoleCLIDownload deployment resource manifest path
-func manifestPath() string {
-	return os.Getenv("CONSOLECLIDOWNLOAD_MANIFEST_PATH")
-}
-
-func replaceKnCLIArtifactsImage(image string, scheme *runtime.Scheme) mf.Transformer {
-	return func(u *unstructured.Unstructured) error {
-		if u.GetKind() == "Deployment" {
-			deploy := &appsv1.Deployment{}
-			if err := scheme.Convert(u, deploy, nil); err != nil {
-				return fmt.Errorf("failed to convert unstructured obj to Deployment: %w", err)
-			}
-
-			containers := deploy.Spec.Template.Spec.Containers
-			for i, container := range containers {
-				if container.Name == knDownloadServer && container.Image != image {
-					log.Info("Replacing", "deployment", container.Name, "image", image)
-					containers[i].Image = image
-					break
-				}
-			}
-
-			if err := scheme.Convert(deploy, u, nil); err != nil {
-				return fmt.Errorf("failed to convert Deployment obj to unstructured: %w", err)
-			}
+// makeKnService makes Knative Service object and its SPEC from provided image parameter
+func makeKnService(image string, instance *servingv1alpha1.KnativeServing) *servingv1.Service {
+	anno := make(map[string]string)
+	if instance != nil {
+		anno = map[string]string{
+			common.ServingOwnerName:      instance.Name,
+			common.ServingOwnerNamespace: instance.Namespace,
 		}
```

Comment:
I don't think we need these as we can just set an OwnerReference to the KnativeServing instance (we're in the same namespace per below)

Reply:
Per our chat, once switched to OwnerReference approach there's a race condition issue with control-plane deletion that causes route's finalizer to hang and block clean resouce removal. I've added a code comment with the explanation.